interpolation.go

```diff
@@ -110,9 +110,9 @@ func InterP1(xList, vList []float64, queryList []float64) ([]float64, error) {
 
 	// Split
 	sortedX, sortedV := make([]float64, size), make([]float64, size)
-	for _, v := range p {
```

Comment:
I thought append still worked here? The capacity is set to the expected size but the Len is still 0 so append should still insert values into the expected indices

Reply:
The length is `size` by doing `make([]float64, size)`, so `append` would append value starting from `size` index.

Reply:
`make([]float64, 0, size)` would do 0 length and `size` capacity